internal/subscriber/subscriber.go

```diff
@@ -571,7 +571,37 @@ func (s *Subscriber) pull(req *PullRequest) {
 				continue
 			}
 			ackID := ackMessage.BuildAckID()
-			sm = append(sm, &metrov1.ReceivedMessage{AckId: ackID, Message: protoMsg, DeliveryAttempt: msg.CurrentRetryCount + 1})
+			// check if the subscription has any filter applied and check if the message satisfies it if any
+
+			isElligible := true
+
+			if s.subscription.FilterExpression != "" {
+				subFilter, err := s.subscription.GetFilterExpressionAsStruct()
```

Comment:
Sub filter can be extracted in the begining when the subscriber starts. Not in every `pull` call

Reply:
To do that, we would then have to create a private field in subscriber to hold the filter_struct, which I think doesn't make sense as filter is a field of subscription and not subscriber or pass along the value in function as a parameter which isn't a better way either. 
Also this extraction is just an access call to a variable. It shouldn't be that costly operation. 

Do let me know if there is other way and added benefit of doing it the other way